be/src/exprs/agg/percentile_cont.h

```diff
@@ -137,34 +164,59 @@ class PercentileContAggregateFunction final
         } else if constexpr (lt_is_arithmetic<LT>) {
             result = new_vector[index] + (u - (double)index) * (new_vector[index + 1] - new_vector[index]);
         } else {
-            LOG(ERROR) << "Invalid LogicalTypes for percentile_cont function";
-            return;
+            // won't go there if percentile_cont is registered correctly
+            throw std::runtime_error("Invalid PrimitiveTypes for percentile_cont function");
         }
 
         column->append(result);
     }
 
-    void convert_to_serialize_format(FunctionContext* ctx, const Columns& src, size_t chunk_size,
-                                     ColumnPtr* dst) const override {
-        if (chunk_size <= 0) {
+    std::string get_name() const override { return "percentile_cont"; }
+};
+
+template <LogicalType LT>
+class PercentileDiscAggregateFunction final : public PercentileContDiscAggregateFunction<LT> {
+    using InputCppType = RunTimeCppType<LT>;
+    using InputColumnType = RunTimeColumnType<LT>;
+    static constexpr auto ResultLT = PercentileResultLT<LT, false>;
+    using ResultType = RunTimeCppType<ResultLT>;
+    using ResultColumnType = RunTimeColumnType<ResultLT>;
+
+    void finalize_to_column(FunctionContext* ctx, ConstAggDataPtr __restrict state, Column* to) const override {
+        using CppType = RunTimeCppType<LT>;
+        std::vector<CppType> new_vector = this->data(state).items;
+        pdqsort(false, new_vector.begin(), new_vector.end());
+        const double& rate = this->data(state).rate;
+
+        ResultColumnType* column = down_cast<ResultColumnType*>(to);
+        if (new_vector.size() == 0) {
             return;
         }
-        auto* dst_column = down_cast<BinaryColumn*>((*dst).get());
-        Bytes& bytes = dst_column->get_bytes();
-        double rate = ColumnHelper::get_const_value<TYPE_DOUBLE>(src[1]);
-        InputColumnType src_column = *down_cast<const InputColumnType*>(src[0].get());
-        InputCppType* src_data = src_column.get_data().data();
-        for (auto i = 0; i < chunk_size; ++i) {
-            size_t old_size = bytes.size();
-            bytes.resize(old_size + sizeof(double) + sizeof(size_t) + sizeof(InputCppType));
-            memcpy(bytes.data() + old_size, &rate, sizeof(double));
-            *reinterpret_cast<size_t*>(bytes.data() + old_size + sizeof(double)) = 1UL;
-            memcpy(bytes.data() + old_size + sizeof(double) + sizeof(size_t), &src_data[i], sizeof(InputCppType));
-            dst_column->get_offset().push_back(bytes.size());
+        if (new_vector.size() == 1 || rate == 1) {
```

Comment:
so for percentile_disc(col, 0) and percentile_disc(col,1), suggest to transformed to min(col) and max(col),   min/max is superior to percentile_disc